### PR TITLE
Resolve 94: Iso3166Countries should have a method for validating a country code

### LINKED ISSUE #94: Iso3166Countries should have a method for validating a country code
The functionality can already be achieved with `var isValid = countries.Get(countryCode) != default(Iso3166Country)`, but that is not very semantic.

Maybe adding something like `var isValid = countries.IsValidCountryCode(countryCode)` would be a useful addition.

The reason that I did not add it originally, was that in my use cases I would always need the country afterwards, in which case the null check made sense.

### PR DESCRIPTION
**Pull request by pingvinen**
This adds `bool IsValid**Code(string)` to the country and currency code classes. The issue only mentions adding it for countries, but these two classes are so similar that I thought it best to add it to the currency class as well.

During testing I also noticed that we did not have any tests checking how the `Get(string)` methods handled null, empty strings, invalid codes etc. so I also added those - and fixed a bug when they were given null.

TL;DR; New methods:
- `bool Iso3166Countries.IsValidCountryCode(string)`
- `bool Iso4217Currencies.IsValidCurrencyCode(string)`

Closes #94

### CODE REVIEW

**Comment by pingvinen** [comment]
Actually, I forgot to add documentation... tsk tsk...

**Comment by pingvinen** [comment]
Ok, now it has documentation as well :) Go ahead and review.

**Review by nover — DISMISSED** [review]

**Comment by nover** [comment]
@pingvinen rebase...

**Review by nover — APPROVED** [review]
